### PR TITLE
Separate schemas refactoring

### PR DESCRIPTION
**Pull request by mehdi-aouadi**
## PR Description
On top of #8569
Refactoring the attester slashing schemas initialisation 

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: acceptance-tests/src/testFixtures/java/tech/pegasys/teku/test/acceptance/dsl/TekuBeaconNode.java [hunk 1]
```diff
@@ -258,7 +259,10 @@ public void postAttesterSlashing(
     final String body =
         JsonUtil.serialize(
             attesterSlashing,
-            spec.getGenesisSchemaDefinitions().getAttesterSlashingSchema().getJsonTypeDefinition());
+            spec.getGenesisSchemaDefinitions()
+                .getAttesterSlashingSchema()
+                .castTypeToAttesterSlashingSchema()
```
Comment (rolfyone) [inline acceptance-tests/src/testFixtures/java/tech/pegasys/teku/test/acceptance/dsl/TekuBeaconNode.java]: id really like to have this less crazy... we really dont want to have to call casts like this everywhere, it'll explode into insanity if we do this for a lot of types..
Why can't we do the `.required` type behaviour?

Comment (mehdi-aouadi) [reply acceptance-tests/src/testFixtures/java/tech/pegasys/teku/test/acceptance/dsl/TekuBeaconNode.java]: I tried to completely get rid of the casts but it didn't go well (ended up with broad wildcards and doing the cast in the tests anyway). Not sure the `required` type is different from what's we're doing tbh. I think it's a broader issue and the schema definition design must be reviewed in general but that's out of the scope of this PR

**Review by tbenr — APPROVED** [review]
LGTM

File: acceptance-tests/src/testFixtures/java/tech/pegasys/teku/test/acceptance/dsl/TekuBeaconNode.java (hunk 1 above)
Comment (tbenr) [reply acceptance-tests/src/testFixtures/java/tech/pegasys/teku/test/acceptance/dsl/TekuBeaconNode.java]: ye it is a bit complicated to sort this out. I vote for merge this one and I hope I can clean this up when I revisit the whole schema handling.

Comment (tbenr) [reply acceptance-tests/src/testFixtures/java/tech/pegasys/teku/test/acceptance/dsl/TekuBeaconNode.java]: I found a way to improve it and remove some of the castType calls

Comment (mehdi-aouadi) [reply acceptance-tests/src/testFixtures/java/tech/pegasys/teku/test/acceptance/dsl/TekuBeaconNode.java]: Definitely cleaner this way, thanks!

**Pull request merged**